pkg/runs/junitReporter.go

```diff
@@ -112,3 +115,27 @@ func ReportJunit(
 	}
 	return err
 }
+
+func sortFinishedRuns(finishedRuns map[string]*TestRun) map[string]*TestRun {
```

Comment:
This makes no sense. Sorting a map, and returning a map. Maps are un-sorted by definition.
If you want a sorted structure, one that retains an order, then you should use a list ? or access the map using the keys, and sort that list of keys ?

Reply:
eg: https://stackoverflow.com/questions/18342784/how-to-iterate-through-a-map-in-golang-in-order

Reply:
Or you could use orderedmap instead of map. See https://pkg.go.dev/github.com/wk8/go-ordered-map#section-readme